### PR TITLE
Fixes #7315: Add migration scripts to authentication providers configuration

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/7315

### CODE REVIEW

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 1]
```diff
@@ -397,6 +398,59 @@ rudder.cfengine.reload.server.command=/opt/rudder/bin/rudder-reload-cf-serverd"
...
+
+
+  # - 3.2.0 : Add properties to configure authentication plugins and master admin
+  # note: we can't use rudder.auth.admin.login (or .password) because it's commented
+  # out by default, so it would defeat the regex. 
+  # We also need to comment out rudder.auth.ldap.enable and use its value to decide 
+  # if we should use "ldap" of "file" for rudder.auth.type 
+
+  STEP="Add properties to configure authentication plugins and master admin"
+  if grep -iEq "rudder.auth.ldap.enable=\s*false" /opt/rudder/etc/rudder-web.properties; then
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-upgrade]: There must be a ^ at the beginning of the regex or you will match comments too.
And shouldn't there be a \s* before = too ?
Was the default value true ? If not, you should reverse the test to make the default 'file' and not ldap.

Comment (fanf) [inline rudder-webapp/SOURCES/rudder-upgrade]: Ah, the ^ disappear somehow along the path, thanks for catching it. 
The \s* before "=" can't do any arm, I will add it to.

**Comment by fanf** [comment]
PR updated

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 2]
```diff
@@ -397,6 +398,59 @@ rudder.cfengine.reload.server.command=/opt/rudder/bin/rudder-reload-cf-serverd"
...
+
+
+  # - 3.2.0 : Add properties to configure authentication plugins and master admin
+  # note: we can't use rudder.auth.admin.login (or .password) because it's commented
+  # out by default, so it would defeat the regex. 
+  # We also need to comment out rudder.auth.ldap.enable and use its value to decide 
+  # if we should use "ldap" of "file" for rudder.auth.type 
+
+  STEP="Add properties to configure authentication plugins and master admin"
+  if grep -iEq "^rudder.auth.ldap.enable\s*=\s*false" /opt/rudder/etc/rudder-web.properties; then
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-upgrade]: What about the default value ?
Was this line mandatory or was the default ldap ?

Comment (fanf) [inline rudder-webapp/SOURCES/rudder-upgrade]: the line was mandatory, defaulting to rudder.auth.ldap.enable=false (so to the new "file" authentication provider).

**Comment by peckpeck** [comment]
OK, so it was not possible for the line to be absent.

**Pull request merged**